File: FPRO/curso2425/EjerciciosSimulacro-Repaso/Ejercicio2/Producto.cs
class Producto
{
    private string nombre;
    private int stock;
    private double precio;

    public Producto(string nombre, int stock, double precio)
    {
        this.nombre = nombre;
        this.stock = stock;
        this.precio = precio;
    }

    public int getStock()
    {
        return this.stock;
    }
    public double getPRecio()
    {
        return this.precio;
    }

    public string getNombre()
    {
        return this.nombre;
    }
    public void MostrarDatos()
    {
        Console.WriteLine("nombre " + nombre);
        Console.WriteLine("precio " + precio);
        Console.WriteLine("stock " + stock);
    }
}

File: AD/codigos2526/T5/tienda_ue/src/main/java/com/example/tienda_ue/model/Producto.java
package com.example.tienda_ue.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Table(name = "productos")
public class Producto {

    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private Long id;

    @Column
    private String nombre;

    @Column
    private double precio;

    @Column
    private String descripcion;

    @Column
    private int stock;

    public Producto(String nombre, double precio, String descripcion, int stock) {
        this.nombre = nombre;
        this.precio = precio;
        this.descripcion = descripcion;
        this.stock = stock;
    }
}
